### DISCUSSION TITLE
How to run DynamicHelloWorldExample and RequestReplyExample ? No feedback

### DISCUSSION THREAD

**Discussion by knorr02** [eProsima/Fast-DDS#4908] [Q&A] [answered]
I'm trying to run all Fast-DDS examples locally on my Ubuntu 22.04 machine and so far most of them work and give feedback if a message was send or received.
When i run ./DDSDynamicHelloWorldExample subscriber and ./DDSDynamicHelloWorldExample publisher it does not give me any other feedback than "Starting" and "Subscriber running" / "Publisher 5 samples"
Same goes for RequestReplyExample.

**Comment by JesusPoderoso** [comment, marked as answer]
Hi @knorr02, thanks for using Fast DDS.
As you can see in the [master README](https://github.com/eProsima/Fast-DDS/blob/master/README.md), there are API breaks and master is unstable in some terms until the release of the following Fast DDS version v3.0.0.
One of the biggest features that is introduced in v3.0.0 is a complete refactor of Dynamic types. 
The examples are being refactored too, so their README, code, and outputs are being improved for the next v3.0.0 release. Mind that the DynamicHelloWorldExample may not work as expected until both Dynamic types and examples refactors are finished.

So, to continue with the discussion, it is highly recommended to use the latest version [v2.14.1](https://github.com/eProsima/Fast-DDS/tree/v2.14.1)

On the one hand, the DynamicHelloWorldExample loads a type from the XML file `helloworld_example_type_profile.xml`.
The publisher shares the TypeObject so another participants can discover it.
The subscriber will discover the type and introspecting it will show the received messages from the publisher.

![image](https://github.com/eProsima/Fast-DDS/assets/120394830/59d9a424-87fe-42b6-bb8a-aed4a0395bc0)

On the other hand, the RequestReplyExample illustrates how to achieve a Client/Server communication between two applications using the Fast DDS request-reply mechanism. More information in [Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/request_reply/request_reply.html)

![image](https://github.com/eProsima/Fast-DDS/assets/120394830/9509146c-f341-414f-baaf-535939eba19a)

**Reply by knorr02** [reply]
Thx, i will try v2.14.1 than.

**Reply by truehang** [reply]
I found RequestReplyExample have bugs, sometimes client will block and get no response from server. Just try 5 times to reproduce the problem. Is there any solutions?

**Reply by JesusPoderoso** [reply]
Hi @truehang, I would appreciate if you open a new issue or a discussion to determine the root cause of your issue.
I've checked locally with the latest version v2.14.3 and it works for me, I could not reproduce your issue.
Can you provide more information (Fast DDS version, OS, compiler version, log traces...)?
